server/src/test/java/com/google/sps/AdRowProcessorTest.java

```diff
@@ -1,30 +1,186 @@
 package com.google.sps;
```

Comment:
Unless I missed it I think tests for empty/null strings would be appropriate, test for malformed urls